### DISCUSSION TITLE
The best way to handle exception

### DISCUSSION THREAD

**Discussion by FebriPratama** [Eptagone/Telegram.BotAPI#52] [Q&A]
Hello 
Im using longpolling to get messages, 

net core 3.1 
Telegram.BotAPI version 6.6.1

`        
        public void StartPolling()
        {
            var updates = Bot.GetUpdates<IEnumerable<Update>>();
            while (true)
            {
                if (updates.Any())
                {
                    foreach (var update in updates)
                    {
                        var botInstance = new SidomarBotService(_Token, _configuration);
                        botInstance.OnUpdate(update);
                    }
                    var offset = updates.Last().UpdateId + 1;
                    updates = Bot.GetUpdates<IEnumerable<Update>>(offset);
                }
                else
                {
                    updates = Bot.GetUpdates<IEnumerable<Update>>();
                }
            }
        }
`

Whats the best way to handle exception throws by startpolling error such as :

![Screenshot 2023-04-17 084604](https://user-images.githubusercontent.com/7270604/232358910-2fe414a5-6e88-453b-b2d1-5b4a993031bf.png)

**Comment by Eptagone** [comment]
Hello.

Mmm, i think a try-catch inside your while statement should be enough for exceptions handling during getting updates.
